src/smt/theory_arith.h

```diff
@@ -345,7 +345,7 @@ namespace smt {
             friend class theory_arith;
         public:
             derived_bound(theory_var v, inf_numeral const & val, bound_kind k):bound(v, val, k, false) {}
-            ~derived_bound() override {}
+            virtual ~derived_bound() override {}
```

Comment:
ditto. 
We have many occurrences of destructors that just use override given that this is the new way to, ehm, override.